src/StreetNameRegistry/Municipality/StreetNameCommandHandlerModule.cs

```diff
@@ -81,6 +81,16 @@ public sealed class StreetNameCommandHandlerModule : CommandHandlerModule
                     municipality.RetireStreetName(message.Command.PersistentLocalId);
                 });
 
+            For<CorrectStreetNameRetirement>()
+                .AddSqlStreamStore(getStreamStore, getUnitOfWork, eventMapping, eventSerializer, getSnapshotStore)
+                .AddEventHash<CorrectStreetNameRetirement, Municipality>(getUnitOfWork)
+                .AddProvenance(getUnitOfWork, provenanceFactory)
+                .Handle(async (message, ct) =>
+                {
+                    var municipality = await getMunicipalities().GetAsync(new MunicipalityStreamId(message.Command.MunicipalityId), ct);
+                    municipality.CorrectStreetNameRetirement(message.Command.StreetNameNames, message.Command.PersistentLocalId);
```

Comment:
`StreetNameNames` should be removed as a property from the command and should not be passed as a parameter here.